WordPress/src/main/java/org/wordpress/android/ui/prefs/notifications/NotificationsSettingsFragment.java

```diff
@@ -532,6 +538,22 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
                     AnalyticsTracker.track(AnalyticsTracker.Stat.NOTIFICATION_PENDING_DRAFTS_SETTINGS_DISABLED);
                 }
             }
+        } else if (key.equals(getString(R.string.wp_pref_custom_notification_sound))) {
+            final String defaultPath =
+                    getString(R.string.notification_settings_item_sights_and_sounds_choose_sound_default);
+            final String value = sharedPreferences.getString(key, defaultPath);
+
+            if (value.trim().toLowerCase().startsWith("file://")) {
+                // sound path begins with 'file://` which will lead to FileUriExposedException when used. Revert to
+                //  default and let the user know.
+                AppLog.w(T.NOTIFS, "Notification sound starts with unacceptable scheme: " + value);
+
+                Context context = getActivity();
```

Comment:
What do you think about using the `Application` context rather than the Activity? In this way, if we detect the notification sound is using an unacceptable scheme, then we can always show the Toast. Otherwise, in the _(not so)_ rare case the Fragment is detached, we won't be telling the user anything, when the action was actually triggered by them (so we should always tell them what happened with their request).

The `isAdded()` check would not be necessary in this case as well

Reply:
Makes sense! Done with 2c0bf23.